gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +236,48 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine. Follow this doc:
+  // https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#.
+  private boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    Process process;
```

Comment:
This should be a more local variable declared inside the blocks.

Reply:
Done.